### ISSUE TITLE
Accordion destroying the component inside it

### ISSUE THREAD

**Issue by Git002** [themesberg/flowbite-svelte#1070]
### Describe the bug

I was using the `Accordion` from flowbite and I saw that whatever component I place under the `<AccordionItem>` , it literally destroys that component when clicked on the Accordion in the UI to collapse and then recreates it when expands.

I got to know this issue as the state of my components were losing when I re-click the accordion to expand. Then I have attached the `onMount()` and `ondestroy()` to console log the events and it was very clear to me that it is indeed destroying and re-creating the component on expand/collapse which doesn't makes sense as Accordions are supposed to hide/show stuff in them not destroy it 🥲

### Reproduction

This is my **Accordion.svelte** that accepts a slot inside:
```
<script lang="ts">
  import { AccordionItem, Accordion } from 'flowbite-svelte';
</script>

<Accordion flush multiple>
  <AccordionItem open>
    <span slot="header">Title</span>
    <div slot="arrowup" />
    <div slot="arrowdown" />
    <slot />
  </AccordionItem>
</Accordion>
```

And I am passing my component (Display.svelte) in my **App.svelte** like so:
```
<script lang="ts">
  import Accordion from '$lib/UIComponents/Accordion.svelte'; 
</script>

<Accordion>
  <Display />
</Accordion>
```

Now that Display component that I just passed gets re-rendered, destroyed, and mounted again which should not happen. Or is it that Flowbite's Accordions weren't supposed to take `slots` or `<svelte:components>` and only props inside as such?

### Flowbite version and System Info

```shell
System:
    OS: Linux 6.4 Fedora Linux 38 (Workstation Edition)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 3.44 GB / 7.48 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
Binaries:
    Node: 19.3.0 - /usr/local/bin/node
    npm: 9.6.7 - /usr/local/bin/npm
npmPackages:
    @sveltejs/kit: ^1.0.0 => 1.22.1 
    flowbite-svelte: ^0.37.4 => 0.37.5 
    svelte: ^3.54.0 => 3.59.2 
    vite: ^4.0.0 => 4.4.2
```

**Comment by jjagielka** [comment]
Yes, that's how it's designed. Svelte directive `{#if ... }` does exactly what you've described.

**Comment by Git002** [comment]
> Yes, that's how it's designed. Svelte directive `{#if ... }` does exactly what you've described.

And that's why I've to create my own accordion component that simply sets it's height to 0 when collapsed and then it's height equal to the element's scrollHeight when expanded. This doesn't cause any problems on hide/show and let's me pass any slot/component without re-rendering issues

And I wish flowbite did that too through CSS. No rendering issues and full flexibility. Would be a great if it is implemented...

**Comment by derheld42** [comment]
> > Yes, that's how it's designed. Svelte directive `{#if ... }` does exactly what you've described.
> 
> And that's why I've to create my own accordion component that simply sets it's height to 0 when collapsed and then it's height equal to the element's scrollHeight when expanded. This doesn't cause any problems on hide/show and let's me pass any slot/component without re-rendering issues

Any URLs to your accordion? There might be others who want that behavior too.

**Comment by Git002** [comment]
> Any URLs to your accordion? There might be others who want that behavior too.

@derheld42 you can simply use the svelte variable inside the class directive. Like `<div class={show ? 'flex' : 'none'}>`
Or something like that
Using if blocks is a bad idea as it triggers re-render to the entire component inside it. So yeah, this solution just hides* it not destroy it...

**Comment by derheld42** [comment]
I tracked it down to some behavior of the Frame class. I tweaked the flowbite-svelte code to use div and I continue to have the interactivity I was looking for. I get all of the up sides of the flowbite/flowbite-svelte accordion component and my reactivity contents inside the AccordionItem continue to function. I don't know what I'm missing not using the Frame component, but at least the accordion functionality seems to work :)

If I submitted a pull request that allows the base to be either Frame or div, would that be of interest to the flowbite-svelte crew?

```
diff -up ./flowbite-svelte/dist/accordion/Accordion.svelte.orig ./flowbite-svelte/dist/accordion/Accordion.svelte
--- ./flowbite-svelte/dist/accordion/Accordion.svelte.orig	2024-01-02 21:33:49.574053993 -0800
+++ ./flowbite-svelte/dist/accordion/Accordion.svelte	2024-01-02 21:34:19.174114225 -0800
@@ -20,9 +20,9 @@ let frameClass;
 $: frameClass = twMerge(defaultClass, $$props.class);
 </script>
 
-<Frame {...$$restProps} class={frameClass} color="none">
+<div {...$$restProps} class={frameClass} color="none">
   <slot />
-</Frame>
+</div>
 
 <!--
 @component
```

**Comment by Git002** [comment]
I don't think so, but hey! Who knows?
Just send in a PR request

I faced few more problems with this library so I wrote all the components needed for my app  by myself and deleted flowbite completely.

**Comment by jjagielka** [comment]
@derheld42 - can you please explain what you've achieved with that change from `Frame` to `div`? `AccordionItem` still uses `if` statement, so what is changing?

BTW, the reason that `#if` is used is to trigger the transition - this is how svelte works.

**Comment by jjagielka** [comment]
@Git002 - why don't you report those *few more problems* so we can work on them instead of rewriting the components? You can as well share your solution to those problems so we can merge them in.

**Comment by derheld42** [comment]
> @derheld42 - can you please explain what you've achieved with that change from `Frame` to `div`? `AccordionItem` still uses `if` statement, so what is changing?
> 
> BTW, the reason that `#if` is used is to trigger the transition - this is how svelte works.

The #if didn't seem to be the thing that was causing my issue -- I kept deleting code to try and figure out where the issue was coming from - heh - that's how I isolated it... I'll try to create a simple repro.. stay tuned.

**Comment by jjagielka** [comment]
> The #if didn't seem to be the thing that was causing my issue -- I kept deleting code to try and figure out where the issue was coming from - heh - that's how I isolated it... I'll try to create a simple repro.. stay tuned.

Please make sure you're using the latest version of `flowbite-svelte`: [v0.44.21](https://www.npmjs.com/package/flowbite-svelte)

**Comment by Git002** [comment]
> @Git002 - why don't you report those *few more problems* so we can work on them instead of rewriting the components? You can as well share your solution to those problems so we can merge them in.

@jjagielka First of all I apologise if my tone sounded off. You guys are really doing a good job with flowbite, it's just that I faced problems with accordion and then customising certain components felt too complex. So I dropped flowbite for svelte completely, not because it was bad but I felt like I should rather have a more custom solution for my app.

Now coming to that if block thing,
You're totally correct about if blocks in Svelte but all I'm saying is to not use it here in an accordion as it causes problems when you pass a component inside it.

For example, when you do:
```
{#if Expand}
    <slot/>
{/if}
```
If the value of Expand is `true` then it will render slot otherwise it would just scrap it off the DOM, removing it completely. 

Now for passing a static HTML or say suppose paragraph, it doesn't matter.

However if you pass a component, then it re-renders it just like it did with the paragraph. Now because it is a component, any JS written in it, or I should say the state of it will get destroy.

So on every accordion toggle, that component would re-render destroying and creating itself messing with the states and values contained inside it. 

For example if I have a variable with a value 0 and onclick it becomes 1 then if you click the accordion again, it would reset it's value to 0 which is not something that you want if you want to keep your state which is what I faced.

# Solution:
Use a class inside a div like so:
 `<div class={Expand ? 'flex' : 'hidden'}>`

Now all it will do is retain the same functionality without destroying or re-creating the component.

**Comment by jjagielka** [comment]
> Use a class inside a div like so:
`<div class={Expand ? 'flex' : 'hidden'}>`

That will hide and show the child component but without the animation i.e. transition.

Unfortunately that's a known svelte issue. See the discussion here: https://github.com/sveltejs/svelte/issues/6336
And there's now solution there, but just put in the queue for `one day` in the future...

**Comment by Git002** [comment]
@jjagielka Ahh, I see... Let's hope for the best, man 🤞

**Comment by derheld42** [comment]
> > The #if didn't seem to be the thing that was causing my issue -- I kept deleting code to try and figure out where the issue was coming from - heh - that's how I isolated it... I'll try to create a simple repro.. stay tuned.
> 
> Please make sure you're using the latest version of `flowbite-svelte`: [v0.44.21](https://www.npmjs.com/package/flowbite-svelte)

okay. I verified I was using v0.44.21. I got closer to the root of the problem - it's on my end, but I don't understand it.

I had a $: block that called a function. I changed the function from this:
```
	$: {
		if (...) {
			callFunc1();
		} else if (...) {
		        callFunc2();
		}
	}
```

to this and my reactivity issue disappeared without using the Frame to div tweak I made in a previous comment.
```
	$: {
		if (...) {
			setTimeout(() => callFunc1(), 0);
		} else if (...) {
		        setTimeout(() => callFunc2(),0);
		}
	}
```

**Comment by tylercollier** [comment]
In my case, my slot content is form inputs. When they are removed from the DOM, they are thus not able to be included with form submissions. I'm interested in a non `#if` solution too. I'd settle for no transition/animations.

**Comment by shinokada** [comment]
Please see the [doc](https://flowbite-svelte.com/docs/components/accordion#using-transitiontype=%E2%80%9Cnone%E2%80%9D)

**Comment by shinokada** [comment]
@tylercollier [This doc](https://flowbite-svelte.com/docs/components/accordion#using-svelte%E2%80%99s-snapshot-to-preserve-the-input) for you.
